models/action.go

```diff
@@ -49,14 +49,43 @@ const (
 	ActionDeleteBranch                            // 17
 )
 
+// KeywordMaskType represents the bitmask of types of keywords found in a message.
+type KeywordMaskType int
+
+// Possible bitmask types for keywords that can be found.
+const (
+	KeywordReference KeywordMaskType = 1 << iota // 1 = 1 << 0
+	KeywordReopen                                // 2 = 1 << 1
+	KeywordClose                                 // 4 = 1 << 2
+)
+
+// IssueKeywordsToFind represents a pairing of a pattern to use to find keywords in message and the keywords bitmask value.
+type IssueKeywordsToFind struct {
+	Pattern     *regexp.Regexp
+	KeywordMask KeywordMaskType
+}
+
 var (
 	// Same as Github. See
 	// https://help.github.com/articles/closing-issues-via-commit-messages
 	issueCloseKeywords  = []string{"close", "closes", "closed", "fix", "fixes", "fixed", "resolve", "resolves", "resolved"}
 	issueReopenKeywords = []string{"reopen", "reopens", "reopened"}
 
-	issueCloseKeywordsPat, issueReopenKeywordsPat *regexp.Regexp
-	issueReferenceKeywordsPat                     *regexp.Regexp
+	// populate with details to find keywords for reference, reopen, close
+	issueKeywordsToFind = []*IssueKeywordsToFind{
+		{
+			Pattern:     regexp.MustCompile(issueRefRegexpStr),
+			KeywordMask: KeywordReference,
+		},
+		{
+			Pattern:     regexp.MustCompile(assembleKeywordsPattern(issueReopenKeywords)),
+			KeywordMask: KeywordReopen,
+		},
+		{
+			Pattern:     regexp.MustCompile(assembleKeywordsPattern(issueCloseKeywords)),
```

Comment:
This is a regex... shouldn't this do the trick?
```go
regexp.MustCompile("(close[sd]?|fix|fixe[sd]?|resolve[sd]?)")
```

Reply:
With the addition of the issue reference portion, yes, it could be reduced to a regex.  I didn't change the process of how the regex patterns were built since it seemed that the existing way could easily support things like language translations or user-provided keywords.